hawk/lib/hb_report.rb

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) 2009-2015 Tim Serong <tserong@suse.com>
+# See COPYING for license.
+
+class HbReport
+  # Note: outfile, errfile are based off path passed to generate() -
+  # don't use them prior to a generate run, or you'll get the wrong path.
+  # Lastexit is global (this is freaky/dumb - everything should be based
+  # off path, and callers need to be updated to understand this - this
+  # will happen when we allow multiple hb_report runs, as
+  # hb_reports_controller is what cares about lastexit)
+  attr_reader :path, :outfile, :errfile, :lastexit
+
+  # Call with e.g.:
+  #  ("#{Rails.root}/tmp/pids/hb_report", "/tmp/hb_report-hawk").
+  # pidfile will be set based on 'tmpbase', hb_report will be generated
+  # at 'path'.  If path is nil, generate won't work -- make sure you set
+  # it ASAP, before calling generate or relying on outfile or errfile!
+  def initialize(name = nil)
+    tmpbase = Rails.root.join('tmp', 'pids')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.